### DISCUSSION TITLE
Channel operator that returns count of elements so far.

### DISCUSSION THREAD

**Discussion by siddharthab** [nextflow-io/nextflow#5355] [Ideas]
Currently, the [reduce](https://www.nextflow.io/docs/latest/reference/operator.html#reduce) operator returns the final accumulated value, but a channel of the intermediate values would also be useful. A simplest use case could be to return the count of items processed so far.

This is similar to the `scan` function in the under-preview recursion features, but is based on Groovy closures alone.

I imagine people currently do something like the below, but I am not sure if the increments to num_items needs to be synchronized or if only a single thread ever touches the variable.
```groovy
num_items = 0
ch.map { num_items++ }
```

**Comment by bentsherman** [comment]
There is a scan operator in [nf-boost](https://github.com/bentsherman/nf-boost) if you'd like to try it. At some point I plan to merge it into Nextflow but it might be a while

**Reply by siddharthab** [reply]
Excellent. It's exactly what I was looking for.
